crates/sqlbuiltins/src/validation.rs

```diff
@@ -22,6 +22,9 @@ pub enum ValidationError {
 
     #[error("Format '{format}' not supported by datasource '{datasource}'")]
     FormatNotSupportedByDatasource { format: String, datasource: String },
+
+    #[error("Partitioning is not supported by format '{format}'")]
+    PartitionByNotSupportedByFormat { format: String },
```

Comment:
I'd think that we could have this be an enum that we already have for the format?